File: src/jv/abstractFactory/Hamburgaria.java
package jv.abstractFactory;

import jv.abstractFactory.Estados.GoHamburgaria;
import jv.abstractFactory.Estados.MgHamburgaria;
import jv.abstractFactory.XtudoTipos.XtudoBasico;
import jv.abstractFactory.XtudoTipos.XtudoCompleto;
import jv.abstractFactory.XtudoTipos.XtudoExtra;
import jv.abstractFactory.XtudoTipos.XtudoIngredientesFactory;

public class Hamburgaria {

    public void criarHamburguer(String cidade, String tipo) {
        switch (tipo) {
            case "completo" -> new XtudoCompleto(getIngredientes(cidade)).preparaCompleto();
            case "básico" -> new XtudoBasico(getIngredientes(cidade)).prepararBasico();
            case "extra" -> new XtudoExtra(getIngredientes(cidade)).prepararExtra();
            default -> throw new IllegalStateException("Unexpected value: " + tipo);
        }
    }

    private XtudoIngredientesFactory getIngredientes(String c) {
        if (c.equals("minas")) {
            return new MgHamburgaria();
        } else if (c.equals("goias")) {
            return new GoHamburgaria();
        }
        return null;
    }
}

File: src/abstractFactory/Hamburgaria.kt
package abstractFactory

import abstractFactory.estados.GoHamburgaria
import abstractFactory.estados.MgHamburgaria
import abstractFactory.xTudoTipos.XTudoBasico
import abstractFactory.xTudoTipos.XTudoCompleto
import abstractFactory.xTudoTipos.XTudoExtra
import abstractFactory.xTudoTipos.XTudoIngredientesFactory

/*
Created by Dhionatã on 05/02/2022
*/
class Hamburgaria {

    fun criarHamburguer(cidade: String, tipo: String?) {
        val xtudo = when (tipo) {
            "completo" -> XTudoCompleto(getIngredientes(cidade))
            "básico" -> XTudoBasico(getIngredientes(cidade))
            "extra" -> XTudoExtra(getIngredientes(cidade))
            else -> {
                println("Tipo de hamburguer não encontrado")
                return
            }
        }
        xtudo.preparar()
        println("\n\n-- Hamburguer--\n\n")
        xtudo.exibir()
    }

    private fun getIngredientes(c: String): XTudoIngredientesFactory {
        return if (c == "minas") {
            MgHamburgaria()
        } else if (c == "goias") {
            GoHamburgaria()
        } else {
            MgHamburgaria()
        }
    }
}